src/core/public/keyboard_shortcut/keyboard_shortcut_service.ts

```diff
@@ -0,0 +1,159 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Any modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+import { KeyboardShortcutSetup, KeyboardShortcutStart, ShortcutDefinition } from './types';
+
+/**
+ * @internal
+ * @experimental
+ */
+export class KeyboardShortcutService {
+  private shortcutsMapByKey = new Map<string, ShortcutDefinition[]>();
+  private namespacedIdToKeyLookup = new Map<string, string>();
+
+  public setup(): KeyboardShortcutSetup {
+    return {
+      register: (shortcut) => this.register(shortcut),
+    };
+  }
+
+  public start(): KeyboardShortcutStart {
+    this.startEventListener();
+
+    return {
+      register: (shortcut) => this.register(shortcut),
+      unregister: (shortcut) => this.unregister(shortcut),
+    };
+  }
+
+  public stop() {
+    this.stopEventListener();
+    this.shortcutsMapByKey.clear();
+    this.namespacedIdToKeyLookup.clear();
+  }
+
+  private getNormalizedKey = (str: string): string => str.toLowerCase();
+
+  private getNamespacedId = (shortcut: Pick<ShortcutDefinition, 'id' | 'pluginId'>) =>
+    `${shortcut.id.toLowerCase()}.${shortcut.pluginId.toLowerCase()}`;
+
+  private getEventKeyString = (event: KeyboardEvent): string => {
+    let key = '';
+
+    if (event.ctrlKey) {
+      key += 'ctrl+';
```

Comment:
i think these should be enums or something in a constants file

Reply:
this will be refactored later to handle it better. @natel926 lets leave a comment saying it is temporary and will be refactored later